airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py

```diff
@@ -1144,6 +1152,48 @@ def group_by_stream(records: List[AirbyteRecordMessage]) -> MutableMapping[str,
 
         return result
 
+    @pytest.fixture(name="file_based_connector")
+    def is_file_based_connector(self, connector_metadata: Dict[str, Any]) -> bool:
+        metadata = connector_metadata.get("data", {})
+        return metadata.get("connectorSubtype") == "file" and metadata.get("supportLevel") == "certified"
+
+    @staticmethod
+    def _get_file_extension(file_name: str) -> str:
+        _, file_extension = splitext(file_name)
+        return file_extension
+
+    def _get_actual_file_types(self, records: List[AirbyteRecordMessage]) -> Set[str]:
+        return {self._get_file_extension(record.data.get("_ab_source_file_url", "")) for record in records}
+
+    @staticmethod
+    def _get_unsupported_file_types(config: List[UnsupportedFileTypeConfig]) -> Set[str]:
+        return {t.extension for t in config}
+
+    async def test_all_supported_file_types_present(self, file_based_connector: bool, inputs: BasicReadTestConfig):
+        if not file_based_connector or inputs.file_types.skip_test:
+            reason = (
+                "Skipping the test for supported file types"
+                f"{' as it is only applicable for certified file-based connectors' if not file_based_connector else ''}."
+            )
+            pytest.skip(reason)
+
+        structured_types = {".avro", ".csv", ".jsonl", ".parquet"}
+        unstructured_types = {".pdf", ".doc", ".docx", ".ppt", ".pptx", ".md"}
+
+        if inputs.file_types.unsupported_types:
+            unsupported_file_types = self._get_unsupported_file_types(inputs.file_types.unsupported_types)
+            structured_types.difference_update(unsupported_file_types)
+            unstructured_types.difference_update(unsupported_file_types)
+
+        missing_structured_types = structured_types - self._file_types
+        missing_unstructured_types = unstructured_types - self._file_types
+
+        # all structured and at least one of unstructured supported file types should be present
+        assert not missing_structured_types and len(missing_unstructured_types) != len(unstructured_types), (
+            f"Please make sure you added files with all of supported structured types {structured_types} "
+            f"and at least one with unstructured type {unstructured_types} to the test account."
```

Comment:
Would you mind updating this to 1) list out the ones that are missing, and 2) provide instructions for marking a file type as unsupported?

Reply:
Updated, please take a look